lib/cdo/aws/cloudfront.rb

```diff
@@ -323,6 +318,18 @@ def self.signed_cookies(resource, expiration_date)
         ]
       }.to_json
 
+      # If we are emulating AWS at all, we just return mock cloudfront policy output
+      if CDO.aws_emulated? || CDO.aws_s3_emulated?
+        return {
+          'CloudFront-Policy-Emulated': Base64.encode64(policy).tr('+=/', '-_~')
```

Comment:
It emulates the CloudFront signer policy needed to allow `/restricted/` (signed cookie) access to the bucket. MinIO nor LocalStack actually emulate CloudFront signing... but it is pretty easy to pretend to do it anyway. This does that.

This policy is checked when the `/restricted/` asset path gets a request and it should have a cookie with the right policy info. It just needs to match... so we can just spit it out as though it happened and it will still more or less effectively emulate signed bucket access.

It allows for this check to succeed: https://github.com/code-dot-org/code-dot-org/pull/60276/files/0965896c56ff2b421b95f073b5f6845f5fc5fe6e#diff-10cfca81b682d8a8836e2f5bf0ea04f83264368ad0df41214ea8ebd53be18ac9R86